DissapearedNumsInArray.java

```diff
@@ -0,0 +1,60 @@
+// Time Complexity : O(n)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : could get with O(n) extra space but didn't get the intuition for O(1)
```

Comment:
The Constant Space solution is based on the constraints that array has only numbers from 1 to n. This condition leads us to mark the respective indices as negative (if the number is present). Thus the algorithm is based on constraints.